### DISCUSSION TITLE
Timestamps before start of Unix epoch

### DISCUSSION THREAD

**Discussion by this-user** [questdb/questdb#2525] [Ideas]
I have been trying to import some data with timestamps before 1970-01-01, and it seems this does not work even though the documentation mentions that timestamps are stored as signed offset from start of the Unix epoch. Do I understand it correctly that storing that kind of data is not currently supported, and that this is the expected behaviour?

**Comment by jerrinot** [comment]
hi @this-user (cool username!:)), you are right. this is currently not supported. see https://github.com/questdb/questdb/issues/1884

**Comment by this-user** [comment]
Thanks for the information. Unfortunately, this is a deal breaker, because I am dealing with time series of financial assets (stocks, indices etc.), quite a few of which go back a lot further. Maybe this is something you would consider adding in the future, because most other TSDBMS either have poor performance when querying data or lack support for use with programming languages other than Python, especially JDBC compatibility.

**Comment by tris0laris** [comment]
Hi, thanks for sharing this. How far back do you need to go?

**Comment by this-user** [comment]
The S&P 500 index has existed since 1957, the Dow Jones Industrial Average since 1896, although I'm not sure how much of that early data is even available in digital form. There are even older time series for things like currency exchange rates and commodity prices, but those are mostly of interest for academic research. I would say support for timestamps from 1900 or maybe even 1920 should cover > 99% of real world scenarios.

**Comment by brunswyck** [comment]
You should mention this in the docs. Also the error given is not helpful enough. Luckily I knew the 1970 thing could be an issue but a user shouldn't have to guess.
`got error: Failed to serialize value of column 'Date' at row index 0 (Timestamp('1959-01-01 00:00:00')): Timestamp -347155200000000000 is negative. It must be >= 0.  [dc=501008]`

**Comment by rodryquintero** [comment]
I manage to get this to work for date of birth columns. You need to setup the date column to ```timestamp```. I load the values using *InfluxDB Line Protocol*  as nanoseconds.
